tests/check_xcode.sh

```diff
@@ -8,72 +8,98 @@
 # mod-pbxproj should already have been installed (through PIP) before
 # running this script.
 
+has_file() {
+	fgrep -q "path = source/${1};" ${XPROJECT}/project.pbxproj
+}
 
-# Determine path of the current script
-HERE=$(cd `dirname $0` && pwd)
+has_encoded_file() {
+	fgrep "path = source/${1};" ${XPROJECT}/project.pbxproj | fgrep -q "fileEncoding = 4;"
+}
+
+add_new() {
+	local OPTS='--tree <group> --parent source'
+	echo -n "'${1}': "
+	python3 -m pbxproj file "${XPROJECT}" "source/${1}" ${OPTS} --target EndlessSky ${2}
+	# Make sure the file was added with Unicode encoding (fileEncoding = 4;)
+	if ! has_encoded_file ${1}; then
+		local GOOD_STR="isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c"
+		local BAD_STR="isa = PBXFileReference; lastKnownFileType = sourcecode.c"
```

Comment:
The source code file type is always `sourcecode.c.x` where x indicates if it is a header or implementation. All non-library files are in this order, but I could also provide a sed address in case mod-pbxproj ever updates and starts supplying an encoding by default.

Reply:
This part of the script is not a critical part of the game or of the CI infrastructure, this is an (extra) service to automatically generate a patch for developers in the very specific situation that they add new files to the project (or move files around).
The worst that would happen if this part doesn't work is that either no patch gets generated or that the fileEncoding is missing.
For me it is okay to keep it like this until we need to fix it sometime in the future.

The most proper long term fix here would also be to send a bugreport or PR to mod-pbxproj to include the fileEncoding for added files.